### PR TITLE
Use elastic search as storage for Jaeger

### PR DESCRIPTION
**Pull request by shanavazh**
Adding stateful set for elastic search service that will be used in jaeger service as a storage type.
For elastic search the memory is set to 10G to start with, incase required we can increase.
To clean the old trace, spans there is jaeger-es-index-cleaner which is setup as a cron job, the single run just cleans once so we need to run every day. It is schedule to run ever day start. Traces that are older than 3 days are cleared for now.
[ch14462]

### CODE REVIEW

**Comment by shanavazh** [comment]
@arramos84 Please take a look at the elastic stateful set, we have already elastic service running in dev. To keep it separate from the current one have a dedicated elastic search for jaeger.
I have re-used some of the setting suggesting in the elastic K8s setup.

File: K8s/helm/templates/elasticsearch-statefulset.yaml [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+  podManagementPolicy: Parallel
+  replicas: 1
+  revisionHistoryLimit: 10
+  selector:
+    matchLabels:
+      service : es-jaeger
+  serviceName: es-jaeger
+  template:
+    metadata:
+      creationTimestamp: null
```
Comment (arramos84) [inline K8s/helm/templates/elasticsearch-statefulset.yaml:18]: not needed

File: K8s/helm/templates/elasticsearch-statefulset.yaml [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+            runAsUser: 1000
+          volumeMounts:
+            - mountPath: /usr/share/elasticsearch/data
+              name: elasticsearch-data
+  volumeClaimTemplates:
+    - metadata:
+        name: elasticsearch-data
+      spec:
+        accessModes: [ "ReadWriteOnce" ]
+        {{ if not .Values.openshift }}
```
Comment (arramos84) [inline K8s/helm/templates/elasticsearch-statefulset.yaml:59]: Are we doing anything with openshift in the chart? Probably not needed.

Comment (arramos84) [reply K8s/helm/templates/elasticsearch-statefulset.yaml:59]: Actually, nm, I see it in the repo.

**Review by arramos84 — APPROVED** [review]

File: K8s/helm/templates/esindex-cleaner-cron.yaml [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+  namespace: monitoring
+spec:
+  schedule: "0 0 * * *"
+  jobTemplate:
+    spec:
+      template:
+        spec:
+          containers:
+            - name: esindexcleaner
+              image: jaegertracing/jaeger-es-index-cleaner:latest
```
Comment (nandadev) [inline K8s/helm/templates/esindex-cleaner-cron.yaml:14]: We should use jaeger version here (v1.37) that we are using instead of latest

**Review by nandadev — APPROVED** [review]

**Pull request merged**